MyBus/MyBus/Model/Road/RoutePoint.swift

```diff
@@ -34,6 +34,34 @@ class RoutePoint: NSObject {
         }
     }
 
+    static func parseFromGeoGoogle(geoPointJson: JSON) -> RoutePoint? {
+        let geoPoint = RoutePoint()
+        let firstResultJson = geoPointJson["results"][0]
+        let isAddress = firstResultJson["address_components"][0]["types"] == [ "street_number" ]
+        let jsonStatus = geoPointJson["status"].stringValue
+
+        guard isAddress else {
+            return nil
+        }
+
+        switch jsonStatus {
+        case "OK":
+            let originLocation = firstResultJson["geometry"]["location"]
```

Comment:
I was thinking that could be a good idea to move this parsing to the RoutePoint class and add a static factory method to receive a firstResultJson variable and return a RoutePoint object